python/ray/data/dataset.py

```diff
@@ -3643,6 +3640,11 @@ def to_random_access_dataset(
                 to provide ~3000 records / second via ``get_async()``, and
                 ~10000 records / second via ``multiget()``.
         """
+        ctx = DatasetContext.get_current()
+        if ctx.use_streaming_executor:
```

Comment:
Should undeprecate this and change it to use runtime format checks.

Reply:
Ah yeah my bad, undeprecated this